syntax = "proto3";

package warrior;

service War {
  rpc StartupStatus (StartupRequest) returns (StartupResponse) {}
  rpc MissileApproaching (MissileApproachingRequest) returns (Empty) {}
}

message Point {
    int32 x = 1;
    int32 y = 2;
}

message StartupRequest {
  int32 soldier_id = 1;
  int32 N = 2;
}

message StartupResponse {
  int32 soldier_id = 1;
  Point current_position = 2;
}

message MissileApproachingRequest {
  Point target = 1;
  int32 time_to_hit = 2;
  int32 type = 3;
}

message Empty {}